astroquery/nasa_exoplanet_archive/core.py

```diff
@@ -262,15 +321,19 @@ def query_object_async(self, object_name, *, table="exoplanets", get_query_paylo
 
         if "where" in criteria:
             warnings.warn(
-                "Any filters using the 'where' argument are ignored in ``query_object``",
+                "Any filters using the 'where' argument are ignored in ``query_object``. Consider using ``query_criteria`` instead.",
                 InputWarning,
             )
-        criteria["where"] = "{0}hostname='{1}' OR {0}name='{1}'".format(prefix, object_name.strip())
+        if table in ["ps", "pscomppars"]:
+            criteria["where"] = "hostname='{1}' OR {0}name='{1}'".format(prefix, object_name.strip())
+        else:
+            criteria["where"] = "{0}hostname='{1}' OR {0}name='{1}'".format(prefix, object_name.strip())
 
         return self.query_criteria_async(
             table, get_query_payload=get_query_payload, cache=cache, **criteria,
         )
 
+    # This should stay the same for now
```

Comment:
the same as what?

Reply:
That was a leftover comment from my initial review. Removing.